File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/command/simple/WeekendMode.java
package com.happs.ximand.ringcontrol.model.object.command.simple;

public enum WeekendMode {
    MODE_NOT_WORK_ON_WEEKENDS((byte) 0),
    MODE_WORK_ON_SATURDAY((byte) 1),
    MODE_WORK_ON_WEEKENDS((byte) 2);

    private byte modeId;

    WeekendMode(byte mode) {
        this.modeId = mode;
    }

    public static WeekendMode getInstanceForModeId(int modeId) {
        for (WeekendMode weekendMode : WeekendMode.values()) {
            if (weekendMode.getModeId() == modeId) {
                return weekendMode;
            }
        }
        throw new IllegalArgumentException();
    }

    public byte getModeId() {
        return modeId;
    }
}

File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/command/simple/WeekendMode.kt
package com.happs.ximand.ringcontrol.model.`object`.command.simple

import com.happs.ximand.ringcontrol.R

enum class WeekendMode(val modeId: Byte, val descriptionResId: Int) {

    MODE_NOT_WORK_ON_WEEKENDS(0.toByte(), R.string.mode_not_work_on_weekends),
    MODE_WORK_ON_SATURDAY(1.toByte(), R.string.mode_work_on_saturday),
    MODE_WORK_ON_WEEKENDS(2.toByte(), R.string.mode_work_on_weekends);

    companion object {
        fun getInstanceForModeId(modeId: Int): WeekendMode {
            for (weekendMode in values()) {
                if (weekendMode.modeId.toInt() == modeId) {
                    return weekendMode
                }
            }
            throw IllegalArgumentException()
        }
    }

}